migrationtools/migrations.py

```diff
@@ -176,7 +198,28 @@ def prepare_updated_identity_data(
             details['last_mc_reg_on'] = details.get('last_mc_reg_on', None) or kwargs['last_mc_reg_on']
 
         # Nurse data. Only replace with new data if not currently present.
-        # TODO: Nurseconnect data
+        if 'nurseconnect' in kwargs and kwargs['nurseconnect']:
+            nc = details.get('nurseconnect', {})
+            if 'faccode' in kwargs:
+                nc['faccode'] = nc.get('faccode', None) or kwargs['faccode']
+            if 'facname' in kwargs:
+                nc['facname'] = nc.get('facname', None) or kwargs['facname']
+            if 'id_type' in kwargs:
+                nc['id_type'] = nc.get('id_type', None) or kwargs['id_type']
+                if kwargs['id_type'] == 'sa_id':
+                    if 'sa_id_no' in kwargs:
+                        nc['sa_id_no'] = nc.get('sa_id_no', None) or kwargs['sa_id_no']
+                elif kwargs['id_type'] == 'passport':
+                    if 'sa_id_no' in kwargs:
+                        nc['passport_no'] = nc.get('passport_no', None) or kwargs['sa_id_no']
+                    if 'passport_origin' in kwargs:
+                        nc['passport_origin'] = nc.get('passport_origin', None) or kwargs['passport_origin']
+            if 'mom_dob' in kwargs:
+                # duplicate of above mom data, but it is clearer here.
+                nc['dob'] = nc.get('dob', None) or kwargs['mom_dob']
+            if 'redial_sms_sent' in kwargs:
+                nc['redial_sms_sent'] = nc.get('redial_sms_sent', None) or kwargs['redial_sms_sent']
+            details['nurseconnect'] = nc
```

Comment:
Is this the same logic as `prepare_new_identity_data()` for nurse connect records?